### ISSUE TITLE
Set of REUSABLE components for building LAYERED icons

### ISSUE THREAD

**Issue by iang0ld** [Templarian/MaterialDesign#390]
eg.
- [ ] `+` sign in bottom right corner
- [ ] `-` sign in bottom right corner
- [ ] `tick` sign in bottom right corner
- [ ] `x` in bottom right corner
- [ ] `clock` in  in bottom right corner

Also expand standard shapes (so we can put letter and other symbols inside).
e.g.
- [ ] solid landscape
- [ ] and portrait icons
- [ ] stop sign - rotated octagon icon with a flat edge on top (line and solid)
- [ ] triangle and upside triangle (line and solid)
- [ ] certificate style icon (font awesome) - lone and solid

**Comment by JamesCoyle** [comment]
Would be a great feature but I don't really think it's possible. The icons with those variants usually require manual adjustment.

**Comment by ShawnWeaver** [comment]
I agree that some of these would be difficult (or impossible), but there are plenty of existing icons that could benefit from being split for layering. For example, all of the "numeric-\*-box" and "numeric-\*-box-outline" icons could be built from the shape icon and the number icon. I've built a few custom icons by splicing the XAML together for things I've needed (no scaling necessary):
![image](https://cloud.githubusercontent.com/assets/6340045/23951174/cc900c3c-0963-11e7-8c0d-e81756f42295.png)
(I think I pulled the "M" from the Markdown icon. I had to scale and center it, but on its own it should be usable in other shapes.)

It would certainly require special scales for use when layered - the "pencil" icon isn't the same scale as in the "pencil-box" icon, but the version from "pencil-box" can be dropped into other shapes (at least the solids) relatively easily:
![image](https://cloud.githubusercontent.com/assets/6340045/23951804/ba2b9f96-0965-11e7-82b9-84705e61811e.png)

So while some of these suggestions (plus, minus, etc) might be impractical, there are certainly some symbols where it could work very well.
